### ISSUE TITLE
React 17 Support

### ISSUE THREAD

**Issue by benrhere** [stripe/react-stripe-js#140]
### Other information
React 17 is functionally very similar to 16, but I get these warns upon npm install after upgrading to it.

npm WARN @stripe/react-stripe-js@1.1.2 requires a peer of react@^16.8.0 but none is installed. You must install peer dependencies yourself.
npm WARN @stripe/react-stripe-js@1.1.2 requires a peer of react-dom@^16.8.0 but none is installed. You must install peer dependencies yourself.

**Comment by RizeXor** [comment]
`npm i @stripe/react-stripe-js @stripe/stripe-js`

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: client2@0.1.0
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"^17.0.1" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.0" from @stripe/react-stripe-js@1.1.2
npm ERR! node_modules/@stripe/react-stripe-js
npm ERR!   @stripe/react-stripe-js@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /home/rize/.npm/eresolve-report.txt for a full report.
```

Not able to install with yarn or npm

**Comment by Sydwelll** [comment]
UP ! Same here

**Comment by RizeXor** [comment]
> UP ! Same here

This package is dead I suggest using `@stripe/stripe-js` and manually use the API.

**Comment by christopher-stripe** [comment]
We're planning to publish to a new version of React Stripe.js with React 17 as a supported peer dependency, which will resolve this error.

React 17 made changes to the timing of `useEffect` cleanup functions, which React Stripe.js makes use of. So before we publish a new version, we are testing React Stripe.js with a variety of React 16 and React 17 integrations to ensure we won't cause any regressions.

**Comment by dweedon-stripe** [comment]
This was fixed in #147 and released in [v1.2.0](https://github.com/stripe/react-stripe-js/releases/tag/v1.2.0)
